airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/operations/RedshiftSqlOperations.java

```diff
@@ -112,7 +112,7 @@ public boolean isValidData(final JsonNode data) {
 
     // check VARCHAR limits for VARCHAR fields within the SUPER object, if overall object is valid
     if (isValid) {
-      final Map<String, Object> dataMap = Jsons.flatten(data);
+      final Map<String, Object> dataMap = Jsons.flatten(data, true);
```

Comment:
Could you describe how this will affect users who are already running sync with this connector? Will this change how arrays appear in Redshift for them?

Reply:
No change to anything that is currently functioning. This function is used to validate that data fits within the maximum allowed data size. Currently, arrays are capped to the string limit (since they are treated as strings). With this change, they will be able to successfully load larger arrays, as the string limit is enforced on string components only.